semver/_shared.ts

```diff
@@ -185,7 +185,10 @@ export function parseBuild(buildmetadata: string) {
   return buildmetadata.split(".").filter(Boolean);
 }
 
-export function parseNumber(input: string, errorMessage: string) {
+export function parseNumber(input: string | undefined, errorMessage: string) {
+  if (input === undefined) {
+    return 0;
+  }
   const number = Number(input);
```

Comment:
This change seems wrong. Instead, I think we should use non-null assertions for the values we pass to the `input` argument. Even if the `input` is `null`, the function should fail, which is what we want.